### PR TITLE
Move upgraders from python to cpp

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70593

Differential Revision: [D33402543](https://our.internmc.facebook.com/intern/diff/D33402543)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

File: torch/jit/operator_upgraders.py [hunk 1]
```diff
@@ -1,120 +0,0 @@
...
-            return self.true_divide_(other)
-        return self.divide_(other, rounding_mode='trunc')
-
-    # TODO: since TS relies on typecheck comment of mypy
-    # adding type: ignore at specific lines in this function
-    # messes up our type refinement. For now, let's not check
-    # type here.
-    @no_type_check
-    @torch.jit.script
-    def full_0_4(size: List[int], fill_value: Union[int, float], *,
```
Comment (cccclai) [inline torch/jit/operator_upgraders.py:55]: Where are they moved to now?

Comment (tugsbayasgalan) [reply torch/jit/operator_upgraders.py:55]: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/operator_upgraders/upgraders_entry.cpp

File: torch/csrc/jit/operator_upgraders/upgraders_entry.cpp [hunk 2]
```diff
@@ -57,19 +58,39 @@ def full_out_0_4(size:List[int], fill_value:number, *, out:Tensor) -> Tensor:
...
-    auto cu = std::make_shared<CompilationUnit>();
-    cu->define(c10::nullopt, entry.second, nativeResolver(), nullptr);
-    Function& jitFunc = cu->get_function(entry.first);
-    GraphFunction& graphFunction = toGraphFunction(jitFunc);
-    populate_content.insert(std::make_pair(entry.first, graphFunction.graph()));
+    auto upgrader_graph = create_upgrader_graph(entry.first, entry.second);
+    populate_content.insert(std::make_pair(entry.first, upgrader_graph));
   }
-
   populate_upgraders_map(std::forward<UpgraderMap>(populate_content));
```
Comment (gmagogsfm) [inline torch/csrc/jit/operator_upgraders/upgraders_entry.cpp:77]: Why is the `forward` here necessary? It should only be needed for arguments or rvalues. here `populate_content` is neither. What do you think?

**Review by gmagogsfm — APPROVED** [review]
Just some nits.

File: torch/jit/generate_bytecode.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from torch._C import _generate_upgraders_bytecode
+from typing import List
+
+def format_bytecode(table):
+    # given a nested tuples, convert them to nested list
+    def listify(content):
+        if not isinstance(content, tuple):
```
[the remark this answers is not in the document]
Comment (tugsbayasgalan) [reply torch/jit/generate_bytecode.py:7]: This won't trigger. I think i was being paranoid

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33402543).

File: torch/jit/generate_bytecode.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    formatted_table = {}
+    for entry in table:
+        identifier = entry[0]
+        content = entry[1]
+        content = listify(content)
+        formatted_table[identifier] = content
+    return formatted_table
+
+def generate_bytecode() -> List:
```
Comment (cccclai) [inline torch/jit/generate_bytecode.py:19]: nit: can we rename it to something like generate_upgraders_bytecode?

**Pull request closed without merging**